fms_fsdp/utils/train_utils.py

```diff
@@ -78,7 +78,11 @@ def train(
         ddp_stats[2] += 1
 
         if profiler:
-            profiler.step()
+            if cfg.profiler_rank0_only:
```

Comment:
would it be simpler to just set profiler=None when rank0? possibly in `get_profiler` or `main_training` instead of here?

Reply:
Yea that was also my first thought.  I didn't do because I would have to pass extra arg "rank" to "get_profiler" function.
Let me think and maybe that's better.